controller/destination/server_test.go

```diff
@@ -79,6 +79,15 @@ func TestSplitDNSName(t *testing.T) {
 		{"example.com..", []string{}, true},
 		{"..example.com.", []string{}, true},
 		{"foo.example.com", []string{"foo", "example", "com"}, false},
+		{"invalid/character", []string{}, true},
+		{"", []string{}, true},
+		{"ALL-CAPS", []string{"ALL-CAPS"}, false},
+		{"This-dns-label-has-63-characters-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", []string{"This-dns-label-has-63-characters-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}, false},
+		{"0O0", []string{"0O0"}, false},
+		{"-hi", []string{}, true},
```

Comment:
A similar case where the name ends with a hyphen would be good too. Similarly, a case where the name only contains hyphens would be good.